crates/core/tedge_agent/src/file_transfer_server/request_files.rs

```diff
@@ -9,7 +9,15 @@ use camino::Utf8Path;
 use camino::Utf8PathBuf;
 
 use super::error::FileTransferRequestError;
-use super::http_rest::HttpConfig;
+
+#[derive(Clone)]
+pub(super) struct FileTransferDir(Arc<Utf8Path>);
+
+impl FileTransferDir {
+    pub(super) fn new(file_transfer_dir: Utf8PathBuf) -> Self {
+        Self(Arc::from(file_transfer_dir))
+    }
```

Comment:
Good idea to decouple file transfer directories from the rest of the config there.

Reply:
Yeah, as I added more certificate configuration, it became really obvious that the bulk of the tests (which use local, not network-based HTTP testing) needed to provide a bunch of information that the application just didn't care about.